### DISCUSSION TITLE
July 2 - Literate Programming II - hypermedia

### DISCUSSION THREAD

**Discussion by tonyfast** [Quansight/writers-workshop#20] [New class]
# the goal of the workshop

* [begin continue journey into modern literate programming by talking about hypermedia](https://nbviewer.jupyter.org/gist/tonyfast/203fee00e29fadaaa0604e915eb7a8f6) and ways to include [hypermedia into notebooks](https://nbviewer.jupyter.org/gist/tonyfast/203fee00e29fadaaa0604e915eb7a8f6/all-the-displays.md.ipynb)

## teacher's preferred ground rules

* How to use the raise handle feature? Keeping the professor on track..
* I encourage y'all to interrupt me when I'm talking or demo'ing. 
  Sometimes I get distracted by my own special interests and this class if for you all.
* To help take notes, please respond to this discussion item and continue editing your one reply during class. i can see the changes on the discussion board as they happen.

## [calls to action] :+1: :-1:  :smile: :confused:  :rocket:  :heart:  :eyes: :tada: attendance

* place any questions you have in the discussions and we can try to discuss them during the workshop.

* react to this discussion so we can take attendance.

![image](https://user-images.githubusercontent.com/4236275/117461813-fd163180-af1b-11eb-901b-bfdee65d8bdc.png)

### Q1 how's your energy

* :+1: i feel energetic from this week
* :-1: i feel exhausted from this week

## note taking

please use this discussion as a place to take notes. create a comment that you edit throughout the lesson. 

_the perfect is the enemy of the good_

## agenda

0. [:00-:07] chit chat
1. show and tell
1. abbreviated history of hypermedia
2. varieties of mediatypes in notebooks
3. discuss strategies and ways we can combine mediatypes to create enhanced narratives.
  * __prompt__ https://www.oreilly.com/pub/pr/3223
5. homeworks
4. [:50-:60] wrapup

write! right?

**Comment by MarsBarLee** [comment]
Notes:
My hyper-media collage contribution this week
![frickfrock](https://user-images.githubusercontent.com/46167686/124298312-1a393c00-db2a-11eb-8f46-fec76e11cec1.png)

**standard**: "jupyter notebooks are tools, must be reproducible and professional"
**exciting**: "snip snop off the tip top, holding hands wearing a frick frock"

The 'paper' the actually from [Tony's recent Jupyter Notebook,](https://nbviewer.jupyter.org/gist/tonyfast/f53d109fffe1e46b90cd2da8ce322f45) where he printed it out and added pen and marker to create a paper collage. 

![image](https://user-images.githubusercontent.com/46167686/124298876-c7ac4f80-db2a-11eb-8593-12eb8ae2c529.png)
It was made by a photomanipluation method known as 'clipping'.

So it went from digital notebook -> printed paper -> paper collage -> digital collage -> meme. Coming back full circle!

Other notes:
Sites that change the visual information as you scroll/interact
This is made with D3, a Javascript based library. With this notebook-based widgets, we could keep it written in python even though lots of widgets use D3, we don't need to write Javascript.
http://www.r2d3.us/visual-intro-to-machine-learning-part-1/

**Reply by willwray** [reply]
The one with 'lang' on it's head and 'bplo' on its boddice is best.
The one on the left will become Darth Frocker

**Reply by willwray** [reply]
Tony's paper has a Scottish feel to it - tarten for the kilts

**Reply by MarsBarLee** [reply]
@willwray You've inspired me... Now I'm thinking of animating this in After Effects to make them dance!
